src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/MauiPlatformTargetFrameworkUpgradeStep.cs

```diff
@@ -126,15 +126,19 @@ protected async override Task<UpgradeStepInitializeResult> InitializeImplAsync(I
 
         private static TargetFrameworkMoniker GetExpectedTargetFramework(string componentFlagProperty)
         {
-            var propertyValue = (ProjectComponents)Enum.Parse(typeof(ProjectComponents), componentFlagProperty);
-            if (propertyValue.HasFlag(ProjectComponents.XamarinAndroid))
+            if (Enum.TryParse<ProjectComponents>(componentFlagProperty, out var propertyValue))
             {
-                return TargetFrameworkMoniker.Net70_Android;
-            }
-            else
-            {
-                return TargetFrameworkMoniker.Net70_iOS;
+                if (propertyValue.HasFlag(ProjectComponents.XamarinAndroid))
+                {
+                    return TargetFrameworkMoniker.Net70_Android;
+                }
+                else
+                {
+                    return TargetFrameworkMoniker.Net70_iOS;
+                }
             }
+
+            throw new ArgumentException(componentFlagProperty);
```

Comment:
How about we return null here instead and then we can return UpgradeStepStatus.Failed similar to line 73?